File: Python3/Medium/WrongAnswer/k-divisible-elements-subarrays_0.py
class Solution:
    def countDistinct(self, nums: List[int], k: int, p: int) -> int:
        
        count = 0
        for i in range(len(nums)):
            for j in range(i, min(i + k, len(nums))):
                if sum(nums[i:j+1]) % p == 0:
                    count += 1
        return count

File: Rust/Medium/CompileError/k-divisible-elements-subarrays_0.rs
impl Solution {
    pub fn count_distinct(nums: Vec<i32>, k: i32, p: i32) -> i32 {
        
        let n = nums.len();
        let mut res = 0;
        let mut pre = vec![0; n + 1];
        for i in 0..n {
            pre[i + 1] = (pre[i] + nums[i]) % p;
        }
        let mut map = std::collections::HashMap::new();
        for i in 0..=n {
            if i as i32 > k {
                *map.entry(pre[i - k as usize - 1]).or_insert(0) -= 1;
            }
            res += map.entry(pre[i]).or_insert(0);
            *map.entry(pre[i]).or_insert(0) += 1;
        }
        res
    }
}